### ISSUE TITLE
MAX17048 and Feather ESP32-S3 I2C

### ISSUE THREAD

**Issue by mikeysklar** [adafruit/Adafruit_MAX1704X#3]
A [forum user reported](https://forums.adafruit.com/viewtopic.php?t=204295) that their Feather ESP32-S3 (4MB Flash / 2MB PSRAM) worked with the MAX1704X_basic.ino demo code until a SHTXX temperature sensor was also on the I2C bus. 

I was able to verify with my ESP32-S3 (Rev TFT) with a SHT41 sensor had the same effect. Same issue with the SGP40 sensor being on the bus. No issue with the VCNL4040.

```
Adafruit MAX17048 simple demo
Couldnt find Adafruit MAX17048?
Make sure a battery is plugged in!
```

Modifying the example code everything works fine after three passes.

change this:

```
  if (!maxlipo.begin()) {
    Serial.println(F("Couldnt find Adafruit MAX17048?\nMake sure a battery is plugged in!"));
    while (1) delay(10);
  }
```

to this:
```
  while (!maxlipo.begin()) {
    Serial.println(F("Couldnt find Adafruit MAX17048?\nMake sure a battery is plugged in!\n"));
    delay(5000);
  }
```

long timeout and multiple tries...once initialized it seems run fine without issue. 

```
Make sure a battery is plugged in!

Couldnt find Adafruit MAX17048?
Make sure a battery is plugged in!

Found MAX17048 with Chip ID: 0xC
Batt Voltage: 0.000 V
Batt Percent: 0.0 %

Batt Voltage: 4.193 V
Batt Percent: 103.7 %
```

**Comment by caternuson** [comment]
The CP library has a loop in its reset method:
https://github.com/adafruit/Adafruit_CircuitPython_MAX1704x/blob/ed44d569cac6ef0b243f43a92214ed81b4ccb808/adafruit_max1704x.py#L127-L134

Instead of looping in user sketch, wonder if modifying this lib's reset would help?

Can you test this modified reset?
```cpp
bool Adafruit_MAX17048::reset(void) {
  Adafruit_BusIO_Register cmd =
      Adafruit_BusIO_Register(i2c_dev, MAX1704X_CMD_REG, 2, MSBFIRST);

  // the actual reset, this should *fail*
  if (cmd.write(0x5400)) {
    return false;
  }

  // loop and attempt to clear alert until success
  for (uint8_t retries=0; retries<3; retries++) {
    if (clearAlertFlag(MAX1704X_ALERTFLAG_RESET_INDICATOR)) {
      return true;
    }
    // add retry delay?
  }

  // something didn't work :(
  return false;
}
```

**Comment by mikeysklar** [comment]
Yes, I can confirm that changing out the MAX17048 libraries reset function with the one you provided works smoothly with the ESP32-S3 (Feather REV TFT) and a SHT41 sensor. This is using the MAX17048_basic.ino with no modifications. 

Thank you for the super fast patch.

**Comment by caternuson** [comment]
Cool. Thanks for testing and confirming. Want to make a PR with that change? No worries if not, I can make one.

**Comment by mikeysklar** [comment]
Sure thing. I'll do the PR in the next hour.

**Comment by mikeysklar** [comment]
The builtin test platform scripts that run after the PR failed. I don't see anything relevant. It does not appear to be related to this code change.

```
########################################
INSTALLING ARDUINO BOARDS
########################################
arduino-cli core update-index --additional-urls https://adafruit.github.io/arduino-board-index/package_adafruit_index.json,http://arduino.esp8266.com/stable/package_esp8266com_index.json,https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_dev_index.json,https://sandeepmistry.github.io/arduino-nRF5/package_nRF5_boards_index.json,https://github.com/earlephilhower/arduino-pico/releases/download/global/package_rp2040_index.json,http://drazzy.com/package_drazzy.com_index.json > /dev/null
attempt 1 failed, 2 retry left
attempt 2 failed, 1 retry left
attempt 3 failed, 0 retry left
FAILED to update core indices
Error: Process completed with exit code [25](https://github.com/adafruit/Adafruit_MAX1704X/actions/runs/6152047243/job/16693385920?pr=4#step:6:26)5.
```

**Comment by caternuson** [comment]
[Commented in the PR](https://github.com/adafruit/Adafruit_MAX1704X/pull/4#issuecomment-1714810323)

**Comment by tyeth** [comment]
Closing as this appears resolved, at least the PR ( #4 ) is merged
